packages/zoe/src/zoeService/escrowStorage.js

```diff
@@ -20,9 +20,24 @@ export const makeEscrowStorage = () => {
   const brandToPurse = makeWeakStore('brand');
 
   /** @type {CreatePurse} */
-  const createPurse = (issuer, brand) => {
+  const createPurse = async (issuer, brand) => {
+    if (brandToPurse.has(brand)) {
+      return;
+    }
+    let purseP;
+    try {
+      purseP = await E(issuer).makeEmptyPurse();
+    } catch (err) {
+      const cannotCreateError = Error(
+        // @ts-ignore Types should allow for a DetailsToken
+        X`A purse could not be created for brand ${brand}`,
+      );
+      assert.note(cannotCreateError, X`Caused by: ${err}`);
+      throw cannotCreateError;
```

Comment:
Unfortunately, we're not in a position to modify the `Error` constructors to understand error details (`X`). In this case, the static type error indicated an actual problem! Instead, use `assert.error`

```suggestion
      const cannotCreateError = assert.error(
        X`A purse could not be created for brand ${brand}`,
      );
      assert.note(cannotCreateError, X`Caused by: ${err}`);
      throw cannotCreateError;
```
or, in this case, consider using `assert.fail` directly. I don't see any reason to hide the underlying `err` cause to unprivileged viewers of the outer error.
```suggestion
      assert.fail(
        X`A purse could not be created for brand ${brand} because: ${err}`,
      );
```